cmd/query/app/handler_otlp_translator_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc"
+)
+
+func readOTLPTraces() (interface{}, error) {
+	dat, err := os.ReadFile("./fixture/OTLP-transform.json")
+	if err != nil {
+		return nil, err
+	} else {
+		var out interface{}
+		json.Unmarshal(dat, out)
+		return out, nil
+
+	}
+}
+
+func readJaegerTraces() (interface{}, error) {
+	dat, err := os.ReadFile("./fixture/jaeger-transform.json")
+	if err != nil {
+		return nil, err
+	} else {
+		var out interface{}
+		json.Unmarshal(dat, out)
+		return out, nil
+	}
+}
+
+func TestTransformOTLPSuccess(t *testing.T) {
+	// mockReader := &spanstoremocks.Reader{}
+	withTestServer(func(ts *testServer) {
+		// make main reader return NotFound
+		var response interface{}
+		request, err := readOTLPTraces()
+		require.NoError(t, err)
+		err = postJSON(ts.server.URL+"/api/transform", request, response)
```

Comment:
I think there is unnecessary JSON marshalling happening in this test. There's definitely no need in trying to parse the input file, just submit the bytes from the file as is. For the output file you may want to parse and re-serialize with  a predefined indentation, so that you can compare it with the fixture.

Reply:
Initially the test I wrote was doing this, but was failing. Later I found out that `postJSON` (that's being used in the test) converts the given "JSON" into a stream of bytes itself. If it converts a stream of bytes into a stream of bytes (thinking its s JSON type) the functions give an error 

https://github.com/jaegertracing/jaeger/blob/e08f576fd64a992ef0396112bc8401472cc9dd92/cmd/query/app/http_handler_test.go#L837-L842

Reply:
yes, but you don't need to use postJSON

Reply:
using `execJSON` and making the http.request manually with the buffer?